### PR TITLE
Character Material generation

### PR DESCRIPTION
**Pull request by priolette**
## Describe your changes

Added character ascension and talent material generation

## Issue or discord link

- https://discord.com/channels/785153694478893126/1259536645342888009/1259612393193865366

## Testing/validation

Check the `allCharacterMat_gen.json` file

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/gi/mats/tsconfig.json
Comment (nguyentvan7) [inline libs/gi/mats/tsconfig.json:1]: Can you add a tsconfig.eslint.json similar to this?
https://github.com/frzyc/genshin-optimizer/blob/master/libs/gi/assets/tsconfig.eslint.json

File: libs/gi/dm/src/dm/character/AvatarPromoteExcelConfigData.ts [hunk 1]
```diff
@@ -54,19 +57,38 @@ type AvatarPromoteExcelConfigData = {
...
+
+export type AscensionRecord = {
+  props: { [key: string]: number }
+  scoinCost: number
+  costItems: Array<
+    | {
+        id: number
+        count: number
+      }
+    | Record<string, never>
```
Comment (nguyentvan7) [inline libs/gi/dm/src/dm/character/AvatarPromoteExcelConfigData.ts:69]: whats the point of this?

File: libs/gi/dm/src/dm/character/ProudSkillExcelConfigData.ts [hunk 2]
```diff
@@ -9,13 +9,14 @@ export type ProudSkillExcelConfigData = {
   descTextMapHash: number //59436107,
   unlockDescTextMapHash: number //4234322242,
   icon: string //"UI_Talent_S_Noel_05",
-  costItems: object[]
+  costItems: Array<{ id: number; count: number }>
```
Comment (nguyentvan7) [inline libs/gi/dm/src/dm/character/ProudSkillExcelConfigData.ts:12]: We tend to use syntax more similar to
```typescript
  costItems: CostItem[]
...
type CostItem = {
  id: number,
  count: number,
}
```

**Review by nguyentvan7 — COMMENTED** [review]
just some questions/style comment

File: libs/gi/dm/src/dm/character/ProudSkillExcelConfigData.ts (hunk 2 above)
Comment (priolette) [reply libs/gi/dm/src/dm/character/ProudSkillExcelConfigData.ts:12]: Thanks, missed this spot

File: libs/gi/dm/src/dm/character/AvatarPromoteExcelConfigData.ts (hunk 1 above)
Comment (priolette) [reply libs/gi/dm/src/dm/character/AvatarPromoteExcelConfigData.ts:69]: That an empty object is possible, not sure what would be the better solution

File: libs/gi/mats/tsconfig.json
Comment (priolette) [reply libs/gi/mats/tsconfig.json:1]: Of course

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**